src/models/booking.cr

```diff
@@ -51,6 +94,9 @@ class Booking
   column created : Int64?
 
   column extension_data : JSON::Any, presence: false
+  column history : Array(History), presence: false
```

Comment:
This should store between 1-3 items. Longes series of events being: reserved, checked_in, checked_out

Reply:
I would add a validation to the model that ensure this and maybe a test to make sure it doesn't save if theres less than 1 or more than 3

Reply:
I have added validation to check that it is no longer than 3.
Allowing a length of 0 should make it compatible with existing bookings that have no history.